commcare_connect/opportunity/visit_import.py

```diff
@@ -311,3 +313,173 @@ def get_status_by_completed_work_id(dataset):
     if invalid_rows:
         raise ImportException(f"{len(invalid_rows)} have errors", invalid_rows)
     return status_by_work_id, reason_by_work_id
+
+
+def bulk_update_catchments(opportunity: Opportunity, file: UploadedFile):
+    file_format = get_file_extension(file)
+    if file_format not in ("csv", "xlsx"):
+        raise ImportException(f"Invalid file format. Only 'CSV' and 'XLSX' are supported. Got {file_format}")
+    imported_data = get_imported_dataset(file, file_format)
+    return _bulk_update_catchments(opportunity, imported_data)
+
+
+class RowDataError(Exception):
```

Comment:
It would be nice to move all exceptions to the top of the file (or maybe even their own `exceptions.py` file) so they are in one place and easy to find.

Reply:
Done